### DISCUSSION TITLE
When is a capsule rebuilt?

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#214] [Q&A] [answered]
- I have a `capsuleA` which uses `capsuleB`.

- I only interact with `capsuleA` at startup by reading it via Container.

- Will `capsuleA` rebuild in the future if `capsuleB` rebuilds, even it there are no listeners for `capsuleA`?

I suppose the answer is yes.

**Comment by GregoryConrad** [comment, marked as answer]
> Will `capsuleA` rebuild in the future if `capsuleB` rebuilds, even it there are no listeners for `capsuleA`?

There are a few possibilities here that dictate what would happen.

First, does `capsuleB` emit a _new_ value (i.e., `newData != oldData`)? If the data returned after a rebuild happens is the same as it was before, then `capsuleA` remains untouched.

Now, say `capsuleB` _does_ emit new data. We must then ask whether `capsuleA` registers any side effects (i.e., `use.something()`)? If it does not register any side effects, then it will be disposed/removed from the container via _idempotent garbage collection_. If it does register a side effect, then it will in fact be rebuilt with the new value from `capsuleB` (because that side effect may rely upon the new value from `capsuleB` in order to maintain _side effect correctness_.

This will all change if the capsule graph gets more complicated/`capsuleA` has downstream dependents. But for just two capsules, the above is correct.

**Reply by GregoryConrad** [reply]
Also, if you're curious, corresponding source code is here: https://github.com/GregoryConrad/rearch-dart/blob/3e97544773e84874d4260f03910f34089d004028/packages/rearch/lib/src/node.dart

**Reply by busslina** [reply]
Good explanation, thanks !!
